### DISCUSSION TITLE
Near contract visibility

### DISCUSSION THREAD

**Discussion by idirall22** [aurora-is-near/rainbow-bridge#597] [Q&A] [answered]
Hello, how can I debug, log and trace the near contracts?

**Comment by birchmd** [comment, marked as answer]
@idirall22 Thanks for kicking off this discussion!

Without a specific example of what you are trying to accomplish I can only given general advice. NEAR contracts have [a logging function](https://docs.rs/near-sdk/3.1.0/near_sdk/env/fn.log.html) exposed as a wasm host function and you can see the logs of a transaction in the block explorer or from the [NEAR CLI](https://docs.near.org/docs/tools/near-cli#near-tx-status).

A good place to start if you're looking for local debugging tools (as opposed to testnet) would be with the [NEAR documentation](https://docs.near.org/docs/develop/contracts/debug-locally).

**Comment by idirall22** [comment]
Thank you, @birchmd this is what I need :).
